mlflow/pmdarima.py

```diff
@@ -0,0 +1,364 @@
+        registered_model_name=registered_model_name,
+        pmdarima_model=pmdarima_model,
+        conda_env=conda_env,
+        signature=signature,
+        input_example=input_example,
+        await_registration_for=await_registration_for,
+        pip_requirements=pip_requirements,
+        extra_pip_requirements=extra_pip_requirements,
+        **kwargs,
+    )
+
+
+def load_model(model_uri, dst_path=None):
+    """
+    Load a ``pmdarima`` ``ARIMA`` model or ``Pipeline`` object from a local file or a run.
+
+    :param model_uri: The location, in URI format, of the MLflow model. For example:
+
+                      - ``/Users/me/path/to/local/model``
+                      - ``relative/path/to/local/model``
+                      - ``s3://my_bucket/path/to/model``
+                      - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+                      - ``mlflow-artifacts:/path/to/model``
+
+                      For more information about supported URI schemes, see
+                      `Referencing Artifacts <https://www.mlflow.org/docs/latest/tracking.html#
+                      artifact-locations>`_.
+    :param dst_path: The local filesystem path to which to download the model artifact.
+                     This directory must already exist. If unspecified, a local output
+                     path will be created.
+
+    :return: A ``pmdarima`` model instance
+    """
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
+    flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
+    pmdarima_model_file_path = os.path.join(
+        local_model_path, flavor_conf.get(_MODEL_BINARY_KEY, _MODEL_BINARY_FILE_NAME)
+    )
+
+    return _load_model(pmdarima_model_file_path)
+
+
+def _save_model(model, path):
+
+    with open(path, "wb") as f:
+        pickle.dump(model, f)
+
+
+def _load_model(path):
+
+    with open(path, "rb") as pickled_model:
+        model = pickle.load(pickled_model)
+    return model
+
+
+def _load_pyfunc(path):
+
+    return _PmdarimaModelWrapper(_load_model(path))
+
+
+class _PmdarimaModelWrapper:
+    def __init__(self, pmdarima_model):
+        import pmdarima
+
+        self.pmdarima_model = pmdarima_model
+        self._pmdarima_version = pmdarima.__version__
+
+    def predict(self, dataframe) -> pd.DataFrame:
+
+        schema = {"n_periods", "X", "return_conf_int", "alpha"}
+        df_schema = dataframe.columns.values.tolist()
+
+        if len(dataframe) > 1:
+            raise MlflowException(
+                f"The provided prediction pd.DataFrame {dataframe.to_string()} "
+                f"contains {len(dataframe)} rows. Only 1 row should be supplied."
+            )
+
+        if not set(df_schema).issubset(schema):
+            raise MlflowException(
+                f"The provided schema {df_schema} contains invalid columns. "
+                f"Columns must be part of: {schema}"
+            )
```

Comment:
The current implementation throws an exception if the given dataframe looks like this:

```python
pd.DataFrame(
    {
        # valid columns
        "n_periods": [0],
        "X": [0],
        "return_conf_int": [0],
        "alpha": [0],
        # invalid columns
        "foo": [0],
    }
)
```

Does the `foo` column cause any issues in the subsequent process?

Reply:
Changed this implementation and simplified the validation check (only n_periods is required so we only throw if we can't get a valid value there). Any extra elements in the passed in DF will be ignored.